### DISCUSSION TITLE
Unusial MHC-profile behavior with special 1DLUT and XYZ->XYZ matrix

### DISCUSSION THREAD

**Discussion by dm-u** [dantmnf/MHC2#17] [Show and tell]
Gradient in paint.net (non-color managed app).

MHC-profile with identity XYZ->XYZ matrix

$$\begin{bmatrix}
1 & 0 & 0 & 0 \\
0 & 1 & 0 & 0 \\
0 & 0 & 1 & 0
\end{bmatrix}$$

and 1DLUT [0, 1], [0.5, 0.5], [0, 1]. Gradient looks as expected:
![PXL_20231215_122138250 (Large)](https://github.com/dantmnf/MHC2/assets/1056017/a930c65c-5383-4c29-9edc-cddbd0f1a78c)

MHC-profile with XYZ->XYZ matrix

$$\begin{bmatrix}
0.99 & 0 & 0 & 0 \\
0 & 0.99 & 0 & 0 \\
0 & 0 & 0.99 & 0
\end{bmatrix}$$

and 1DLUT [0, 1], [0.5, 0.5], [0, 1]. Something strange for low levels:
![PXL_20231215_122231694 (Large)](https://github.com/dantmnf/MHC2/assets/1056017/b08614d1-7f9c-4e30-b47f-38641d7d3d39)

Closer:
![PXL_20231215_122246965 (Large)](https://github.com/dantmnf/MHC2/assets/1056017/c0b55ad4-056b-43da-ab47-cbffd00e9f75)

Any ideas?

**Comment by dantmnf** [comment]
I have also reproduced this with an identity matrix on AMD GPU.

It seems this is driver-defined behavior and varies across GPU vendors.

P.S. Someone also reported that color (0, 0, 0) can't be transformed with MHC2 pipeline on NVIDIA GPU.
